sandbox/rocky/tf/algos/network/actor_critic_net.py

```diff
@@ -0,0 +1,220 @@
+                output = _fc(fc2_out, self._action_dim)
+                output = tf.tanh(output)
+        return output
+
+    def predict(self, state):
+        return self._session.run(
+            self.action, feed_dict={
+                self._state: state,
+            })
+
+    def train(self, state):
+        return self._session.run(
+            [self._train_op], feed_dict={
+                self._state: state,
+            })
+
+    def update_target_net(self):
+        self._session.run(self._update_paras)
+
+    @property
+    def action(self):
+        return self._action
+
+
+class CriticNet(ActorCriticBaseModel):
+    def __init__(self, sess, weight_decay=0.01, gamma=0.99, lr=1e-3):
+        self.name = 'critic'
+
+        self._weight_decay = weight_decay
+        self._gamma = gamma
+        self._lr = lr
+        self._session = sess
+
+    def build_net(self, state, action, reward, terminal, target_q,
+                  action_predict):
+        self._state = state
+        self._action = action
+        self._reward = reward
+        self._target_q = target_q
+        self._terminal = terminal
+        self._action_predict = action_predict
+
+        self._q_value = self._build_net(state, action)
+
+        #for calculating grade of actor, no meaning
+        self._q_value_predict = self._build_net(state, action_predict, True)
+        self._action_loss = -tf.reduce_mean(self._q_value_predict)
+
+        self._get_target_Q(reward, terminal)
+        self._get_loss(self._target_q)
+
+    def _get_target_Q(self, rewards, terminals):
+        self._target_Q = rewards + (
+            1. - terminals) * self._gamma * self._q_value
+
+    def _get_loss(self, target_q):
+        self._loss = tf.reduce_mean(tf.square(target_q - self._q_value))
+
+        critic_reg_vars = [
+            var for var in self.trainable_vars
+            if 'kernel' in var.name and 'output' not in var.name
+        ]
+
+        if self._weight_decay > 0.:
+            vars = [
+                var for var in self.trainable_vars
+                if 'weights' in var.name and 'output' not in var.name
+            ]
+            vars += [
+                var for var in self.trainable_vars
+                if 'kernel' in var.name and 'output' not in var.name
+            ]
+            self._loss += tc.layers.apply_regularization(
+                tc.layers.l2_regularizer(self._weight_decay),
+                weights_list=vars)
+
+        self._optimizer = tf.train.AdamOptimizer(
+            self._lr, beta1=0.9, beta2=0.999, epsilon=1e-08)
+
+        grads = tf.gradients(self._loss, self.trainable_vars)
+        grads_and_vars = list(zip(grads, self.trainable_vars))
+        self._train_op = self._optimizer.apply_gradients(grads_and_vars)
+
+        self._action_grads = tf.gradients(self._action_loss,
+                                          self._action_predict)
+
+    def _build_net(self, state, action, reuse=False):
```

Comment:
add support for user-defined number and sizes of hidden layers, e.g. `hidden_layers=[64, 64]`